collect_app/src/androidTest/java/org/odk/collect/android/espressoutils/pages/Page.java

```diff
@@ -17,6 +17,9 @@
 /**
  * Base class for Page Objects (https://www.martinfowler.com/bliki/PageObject.html)
  * used in Espresso tests. Provides shared helpers/setup.
+ *
+ * This class's generic type is designed to it's extending class so that shared methods
```

Comment:
> it's -> its

Noooooooo. I feel like I spend half my life thinking about this. Will fix the typos.

> How about being a little more explicit and detailed about the pattern

Yeah I agree. Will improve the docs. 

> Seems it would be great to have a This keyword

Haha yeah the fluent pattern really pushes Java's generics here - defining methods on a super class that return an inheriting type. As a fun aside, Kotlin introduced `inline` and `reified` to give you the ability to turn generic types into "real" types that make all this a little easier: https://kotlinlang.org/docs/reference/inline-functions.html#reified-type-parameters